gorqlite_test.go

```diff
@@ -0,0 +1,74 @@
+package gorqlite_test
+
+import (
+	"log"
+	"math/rand"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/rqlite/gorqlite/v2"
+)
+
+var globalConnection *gorqlite.Connection
+
+func TestMain(m *testing.M) {
+	rand.Seed(time.Now().UnixNano())
+
+	conn, err := gorqlite.Open(testUrl())
+	if err != nil {
+		log.Fatalf("opening connection: %v", err)
+	}
+
+	_, err = conn.Write([]string{
+		`CREATE TABLE IF NOT EXISTS ` + testTableName() + ` (id integer, name text)`,
+		`CREATE TABLE IF NOT EXISTS ` + testTableName() + `_full (id integer, name text, wallet real, bankrupt boolean, payload blob, ts DATETIME)`,
+		`CREATE TABLE IF NOT EXISTS ` + testTableName() + `_nullable (id integer, nullstring text, nullint64 integer, nullint32 integer, nullint16 integer, nullfloat64 real, nullbool integer, nulltime integer) strict`,
+	})
+	if err != nil {
+		log.Fatalf("creating table: %v", err)
+	}
+
+	globalConnection = conn
+
+	exitCode := m.Run()
+
+	_, err = conn.Write([]string{
```

Comment:
Since tests can fail and this won't necessarily run to clean up the database if that happens, there are two options:

1- Run this in a `defer` statement and place it before `m.Run()` so that it always gets executed.
2- Keep it here, but also call it at the beginning before creating the tables so that tests start with a fresh/clean database.

The only benefit of the second option is to give the user the ability to inspect the data in the database when tests fail, which I prefer.

Reply:
Note: this won't be necessary if we revert to having each test drop/create its own table and tear it down afterwards.

Reply:
1. If we use `os.Exit()`, any `defer` statement won't be run.
2. I'll do the second option.

Reply:
I guess this is no longer relevant since creating/dropping tables will be moved to the test suites.